1. Basic Web Development/index.html

```diff
@@ -23,63 +22,122 @@
+                <h1>Design your preferences pop-up</h1>
+                <form id="creation-form">
+                    <h2>Cathegory</h2>
+                    <div>
+                        <input id="permissionAll" name="presetC" type="radio" value="All" />
+                        <label for="permissionAll">All</label>
+                    </div>
+                    <div>
+                        <input id="permissionPms" name="presetC" type="radio" value="Permissive" />
+                        <label for="permissionPms">Permissive</label>
                     </div>
-                    <div class="grid-element">
-                        <label for="permissive">Permissive</label>
-                        <input id="permissive" name="preset" type="radio" value="Permissive" />
+                    <div>
+                        <input id="permissionNss" name="presetC" type="radio" value="Necessary" checked />
+                        <label for="permissionNss">Necessary</label>
                     </div>
-                    <div class="grid-element">
-                        <label for="necessary">Necessary</label>
-                        <input id="necessary" name="preset" type="radio" value="Necessary" checked/>
+                    <h2>Permisison text</h2>
+                    <div class="text-input-element">
+                        <label for="permName">Display text</label>
+                        <input id="permName" name="permName" type="text" />
                     </div>
+                    <p id="text-alert" class="no-display">Text is required!</p>
+                    <button type="submit" class="btn-confirm">Confirm</button>
                 </form>
             </div>
-            <div class="section">
-                <h2>Permissions:</h2>
-                <form id="permissions-form">
-                    <div class="grid-element">
-                        <label for="data">Send all your data to Mr Zuck</label>
-                        <input id="data" name="permissionNss" type="checkbox" value="data" checked disabled />
-                    </div>
-                    <div class="grid-element">
-                        <label for="record">Record and store all private interactions</label>
-                        <input id="record" name="permissionNss" type="checkbox" value="record" checked disabled />
-                    </div>
-                    <div class="grid-element">
-                        <label for="harvest">Harvest device specifications</label>
-                        <input id="harvest" name="permissionPms" type="checkbox" value="harvest" />
-                    </div>
-                    <div class="grid-element">
-                        <label for="laugh">Laugh at your poor life choices</label>
-                        <input id="laugh" name="permissionPms" type="checkbox" value="laugh" />
+            <div id="edit-page" class="no-display">
```

Comment:
There is little to no difference between the structure of the create/edit page. Maybe you can do something to use the same structure but detect from the route (query param) on which of the two ones you are in order to prepopulate it on edit and use the correct endpoint?

Reply:
I could write a function that does that but I think it will be way to convoluted as there are about 19 elements for each form that need to be generated. Also, I don't think there is a point in generating the forms when the route changes when I could just generate them in the beginning and toggle their visibility. I could use the query params to set the ids but the function will still require a bunch of if-else statements which will make it even more convoluted. 
I agree with you that the HTML is a bit ugly this way and this section looks like copy-paste code but I believe moving it to ts will make the ts file even uglier. 
I'll still try and see how it ends up.

Reply:
As we discussed in our sync, the elements should stay the same, you only generate one form from where you take your data, but on submit you call the api point based on the current route (add or edit).